src/software/new_geom/polynomial2d.h

```diff
@@ -25,14 +25,40 @@ class Polynomial2d
      */
     Polynomial2d(Polynomial1d poly_x, Polynomial1d poly_y);
 
+    /**
+     * Construct a 2d polynomial that passes through the given points
+     *
+     * The created polynomial will be at the first point at t=0, and at the last point at
+     * t=1, with the other points linearly interpolated in between; ie. if we had three
+     * points, the second point would be at t=0.5
+     *
+     * @param points A series of points the created polynomial will pass through. There
+     *               must be at least two points.
+     * @throw std::invalid_argument If there are less then two points
+     */
+    Polynomial2d(std::vector<Point> points);
```

Comment:
Mark the constructors as `explicit` and take param by `const reference`

Reply:
Just bumping this comment, nothing to add

Reply:
Whoops, fixed